### ISSUE TITLE
How to get all the Marker points in line graph with multiple lines.

### ISSUE THREAD

**Issue by KunalDev** [danielgindi/Charts#4103]
I am implementing line chart with Multiple(Two) lines (Sales of current month and previous month on the day basis). 
On graph click we can pop up a view (BalloonMarker) which shows Graph Point.
But there is a limitation. It shows only one line point.
Is there way we can get points for both lines? And display round circles on selected points on both lines?
I also need to customize BalloonMarker view.

I have attached current and expected layout image:

**Current:-**

<img width="304" alt="Current" src="https://user-images.githubusercontent.com/13748740/62781987-d91f8500-bad6-11e9-842a-0d69e02084e6.png">

**Expected:**

<img width="295" alt="Expected" src="https://user-images.githubusercontent.com/13748740/62782015-ec325500-bad6-11e9-8d21-fecd0d0665b1.png">

**Comment by kkieffer** [comment]
I don't think this is possible because you don't necessarily have the same X coordinate for both data sets.  You're not using the same X range with two different Y values on the line graph.  

However, something that would be nice is to identify in the marker which line graph the marker is referring to.  I don't think this is currently possible.  For example in my chart:

<img width="293" alt="graph" src="https://user-images.githubusercontent.com/33581862/64664847-852df600-d41e-11e9-9c66-3bba3a15ea9c.png">

The BalloonMarker points to the green plot, but the label is just "Value" because I don't know which one it's pointing to.  The IMarker interface func is as follows:

` override func refreshContent(entry: ChartDataEntry, highlight: Highlight)`

The "entry" parameter has x and y values but doesn't provide a way to know which line is currently being pointed to.

**Comment by kkieffer** [comment]
Spoke too soon.  There's an additional "data" field that's an Any? in ChartDataEntry.  When you create your data set you can set this to a string.  Then in refreshContent() function:

`let label = entry.data as! String`

and label is the type of your Y value

**Comment by thierryH91200** [comment]
it's possible 
<img width="2083" alt="Capture" src="https://user-images.githubusercontent.com/6251463/64683004-bb9d5c80-d482-11e9-88c7-636d99196a9f.png">

**Comment by KunalDev** [comment]
@kkieffer Thanks. It helps me.

I am able to display alert for both the points. Passing x value index as “data" field in Any? in ChartDataEntry.
<img width="382" alt="Alertt" src="https://user-images.githubusercontent.com/13748740/65244581-863ee180-db08-11e9-8432-daf1683360b2.png">

**Comment by PrashantKT** [comment]
How to show pop up arrow with custom XIB ?

**Comment by ninevcat** [comment]
> I am implementing line chart with Multiple(Two) lines (Sales of current month and previous month on the day basis).
> On graph click we can pop up a view (BalloonMarker) which shows Graph Point.
> But there is a limitation. It shows only one line point.
> Is there way we can get points for both lines? And display round circles on selected points on both lines?
> I also need to customize BalloonMarker view.
> 
> I have attached current and expected layout image:
> 
> **Current:-**
> 
> <img alt="Current" width="304" src="https://user-images.githubusercontent.com/13748740/62781987-d91f8500-bad6-11e9-842a-0d69e02084e6.png">
> 
> **Expected:**
> 
> <img alt="Expected" width="295" src="https://user-images.githubusercontent.com/13748740/62782015-ec325500-bad6-11e9-8d21-fecd0d0665b1.png">

Have u figure out how to display round circles on selected points on both lines?
![Bitmap@2x](https://user-images.githubusercontent.com/15649946/91831664-f04b8a80-ec76-11ea-9d45-43b0335ef905.png)

**Comment by pu943980293** [comment]
Has the problem been solved? I have the same problem

**Comment by banghuazhao** [comment]
same problem

**Comment by davita00101110dmg** [comment]
> > I am implementing line chart with Multiple(Two) lines (Sales of current month and previous month on the day basis).
> > On graph click we can pop up a view (BalloonMarker) which shows Graph Point.
> > But there is a limitation. It shows only one line point.
> > Is there way we can get points for both lines? And display round circles on selected points on both lines?
> > I also need to customize BalloonMarker view.
> > I have attached current and expected layout image:
> > **Current:-**
> > <img alt="Current" width="304" src="https://user-images.githubusercontent.com/13748740/62781987-d91f8500-bad6-11e9-842a-0d69e02084e6.png">
> > **Expected:**
> > <img alt="Expected" width="295" src="https://user-images.githubusercontent.com/13748740/62782015-ec325500-bad6-11e9-8d21-fecd0d0665b1.png">
> 
> Have u figure out how to display round circles on selected points on both lines? ![Bitmap@2x](https://user-images.githubusercontent.com/15649946/91831894-36085300-ec77-11ea-93f2-4dd5a0261e79.png)

Have you figured out how to outline circles?

**Comment by kkieffer** [comment]
Have a look at [this thread](https://stackoverflow.com/questions/34981568/how-can-i-add-a-marker-on-ios-charts)
